components/TheGrid.vue

```diff
@@ -0,0 +1,305 @@
+<script setup lang="ts">
+import { onMounted, reactive, ref } from 'vue'
+import { useRoute } from 'vue-router'
+import {
+  GridLayout,
+  type Breakpoint,
+  type Breakpoints,
+  type Layout,
+} from 'grid-layout-plus'
+
+import WidgetTitleLink from '@/components/WidgetTitleLink.vue'
+import WidgetImage from '@/components/WidgetImage.vue'
+import WidgetXTimeline from '@/components/WidgetXTimeline.vue'
+import WidgetXPost from '@/components/WidgetXPost.vue'
+import WidgetInstagramPost from '@/components/WidgetInstagramPost.vue'
+import WidgetText from '../components/WidgetText.vue'
+import { type GridProperties, WidgetType, type Widget } from '../types/grid'
+import { getGridConfig, upsertGridConfig } from '../utils/gridConfig'
+import {
+  isValidLayout,
+  getNewUserLayout,
+  SHOWCASE_LAYOUT,
+} from '../utils/gridLayout'
+
+const COL_NUM_LARGE = 4
+const COL_NUM_SMALL = 2
+const ROW_HEIGHT = 20
+
+const cols: Breakpoints = {
+  xxs: COL_NUM_SMALL,
+  xs: COL_NUM_LARGE,
+  sm: COL_NUM_LARGE,
+  md: COL_NUM_LARGE,
+  lg: COL_NUM_LARGE,
+}
+
+const gridOptions = reactive<GridProperties>({
+  isDraggable: false,
+  isResizable: false,
+  isResponsive: true,
+})
+const layout = ref<Widget[]>([])
+const showSettingsModal = ref(false)
+const layoutStringified = ref('')
+
+// Load the layout based on the route id
+const route = useRoute()
+const address = route.path.split('/')[1]
```

Comment:
This is not ideal in case there will be for instance query params in future. To get current address use:

```js
const viewedProfileAddress = getCurrentProfileAddress()
```

Reply:
fixed